pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -96,92 +97,90 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	var created bool
 	var user *userv1.User
-	if user, created, err = r.ensureUser(userAcc); err != nil {
-		_ = r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error())
+	if user, err = r.ensureUser(reqLogger, userAcc); err != nil {
```

Comment:
When I start operator locally and then try to create user_acocunt .. I am getting below error .. I guess this is due to race_condition that we fix before with introducing `created` flag.
```
{"level":"info","ts":1561608964.0394926,"logger":"controller_useraccount","msg":"Reconciling UserAccount","Request.Namespace":"toolchain-member-operator","Request.Name":"johnsmith"}
{"level":"info","ts":1561608964.0395193,"logger":"controller_useraccount","msg":"creating a new user","Request.Namespace":"toolchain-member-operator","Request.Name":"johnsmith","name":"johnsmith"}
{"level":"info","ts":1561608964.050983,"logger":"controller_useraccount","msg":"user created successfully","Request.Namespace":"toolchain-member-operator","Request.Name":"johnsmith","name":"johnsmith"}
{"level":"info","ts":1561608964.0510187,"logger":"controller_useraccount","msg":"creating a new identity","Request.Namespace":"toolchain-member-operator","Request.Name":"johnsmith","name":"rhd:1a03ecac-7c0b-44fc-b66d-12dd7fb21c40"}
{"level":"info","ts":1561608964.0628774,"logger":"controller_useraccount","msg":"identity created successfully","Request.Namespace":"toolchain-member-operator","Request.Name":"johnsmith","name":"rhd:1a03ecac-7c0b-44fc-b66d-12dd7fb21c40"}
{"level":"info","ts":1561608964.077371,"logger":"controller_useraccount","msg":"user-identity mapping created successfully","Request.Namespace":"toolchain-member-operator","Request.Name":"johnsmith","name":"rhd:1a03ecac-7c0b-44fc-b66d-12dd7fb21c40"}
{"level":"info","ts":1561608964.1616054,"logger":"controller_useraccount","msg":"Reconciling UserAccount","Request.Namespace":"","Request.Name":"johnsmith"}
{"level":"info","ts":1561608964.1617534,"logger":"controller_useraccount","msg":"creating a new user","Request.Namespace":"","Request.Name":"johnsmith","name":"johnsmith"}
{"level":"error","ts":1561608964.2006922,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"useraccount-controller","request":"/johnsmith","error":"failed to create user '[johnsmith]': users.user.openshift.io \"johnsmith\" already exists","errorVerbose":"users.user.openshift.io \"johnsmith\" already exists\nfailed to create user '[johnsmith]'\ngithub.com/codeready-toolchain/member-operator/pkg/controller/useraccount.(*ReconcileUserAccount).wrapErrorWithStatusUpdate\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/useraccount/useraccount_controller.go:203\ngithub.com/codeready-toolchain/member-operator/pkg/controller/useraccount.(*ReconcileUserAccount).ensureUser\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/useraccount/useraccount_controller.go:131\ngithub.com/codeready-toolchain/member-operator/pkg/controller/useraccount.(*ReconcileUserAccount).Reconcile\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/useraccount/useraccount_controller.go:101\ngithub.com/codeready-toolchain/member-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215\ngithub.com/codeready-toolchain/member-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1337","stacktrace":"github.com/codeready-toolchain/member-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/codeready-toolchain/member-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\ngithub.com/codeready-toolchain/member-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1561608965.2020411,"logger":"controller_useraccount","msg":"Reconciling UserAccount","Request.Namespace":"","Request.Name":"johnsmith"}
{"level":"info","ts":1561608965.202231,"logger":"controller_useraccount","msg":"user already exists","Request.Namespace":"","Request.Name":"johnsmith","name":"johnsmith"}
{"level":"info","ts":1561608965.2023094,"logger":"controller_useraccount","msg":"identity already exists","Request.Namespace":"","Request.Name":"johnsmith","name":"rhd:1a03ecac-7c0b-44fc-b66d-12dd7fb21c40"}
{"level":"info","ts":1561608965.2174907,"logger":"controller_useraccount","msg":"user-identity mapping already exists","Request.Namespace":"","Request.Name":"johnsmith","name":"rhd:1a03ecac-7c0b-44fc-b66d-12dd7fb21c40"}
```

Reply:
Yes. This is because of the removing the flag. I caught that bug too during writing tests. Will fix it shortly.

Reply:
Fixed + added tests.